### ISSUE TITLE
Filebeat-monitoring should be started first

### ISSUE THREAD

**Issue by AndersonQ** [elastic/elastic-agent#192]
When the elastic-agent is starting the process, it might get stuck trying to start a process and therefore it does not start anything else. If it happens before filebeat-monitoring, there will be no logs available on Elasticsearch. The only way to understand what is happening is ssh into the machine and checking (or extracting) the logs manually. Starting filebeat-monitoring before any other process will improve our ability to diagnose problems.

**Comment by ph** [comment]
Agree, we should change that behavior, without having monitoring on (metric and log) we are indeed blind to support our users from the fleet UI.

**Comment by jlind23** [comment]
@ycombinator is this issue still relevant?

**Comment by ycombinator** [comment]
Looks like this issue was created before we had the ability to request Agent diagnostics from the Fleet UI (https://github.com/elastic/kibana/pull/149575; merged on January 27, 2023).  Given that users can now request Agent diagnostics from the Fleet UI, I don't think this issue is as relevant as before any more. Closing.
